api/rest/restapi.go

```diff
@@ -273,6 +278,12 @@ func (api *API) routes() []route {
 			"/health/graph",
 			api.graphHandler,
 		},
+		{
+			"FilesAdd",
+			"POST",
+			"/files/add",
```

Comment:
call me crazy, but I think this might work `POST /allocations`. As it is effectively creating an Allocation (something to pin) and makes sense that it returns a PinInfo type (possibly)

Reply:
I like this idea too.  I'll do this (leaving off the pininfo for this PR)